src/lib.rs

```diff
@@ -306,6 +315,16 @@ impl Surface {
         self.surface_impl.resize(width, height)
     }
 
+    /// Copies the window contents into a buffer.
+    ///
+    /// ## Platform Dependent Behavior
```

Comment:
It was mentioned in https://github.com/rust-windowing/softbuffer/pull/104#issuecomment-1546966584 that this won't work for "window" contents set from webgl. Sounds like a caveat that should be mentioned here? If the intention is to capture window contents not presented with softbuffer.

Not sure how that works on Windows or X11.

Reply:
It's kinda covered in the documentation here, saying an object "valid to draw a 2D buffer to":
https://github.com/rust-windowing/softbuffer/blob/eb227f10f06f8e5cd4d34ab0c3694c4e60606ff7/src/lib.rs#L226-L227
And throws the appropriate error here:
https://github.com/rust-windowing/softbuffer/blob/eb227f10f06f8e5cd4d34ab0c3694c4e60606ff7/src/web.rs#L68-L70

Though I'm a big fan of https://rust-lang.github.io/rust-clippy/master/index.html#missing_errors_doc and would prefer us to document every possible error on every function.

I made https://github.com/rust-windowing/softbuffer/pull/113 and added some documentation where I felt it was appropriate.